### ISSUE TITLE
[DllImport] Improve pinvoke for `GetDroppedFiles()` and similar.

### ISSUE THREAD

**Issue by jasonswearingen** [ChrisDill/Raylib-cs#97]
Before submitting a new issue, please verify and check:

 - [x ] The issue is specific to Raylib-cs and not raylib
 - [ x] I checked there is no similar issue already reported
 - [ x] My code has no errors or misuse of Raylib-cs

### Issue description

In `raylib.h` the definition is:
```c
RLAPI char **GetDroppedFiles(int *count);  
```

Currently in `raylib-cs` the following is used:
```cs
        [DllImport(nativeLibName, CallingConvention = CallingConvention.Cdecl)]
        public static extern IntPtr GetDroppedFiles(ref int count);
```

I suggest using a call more precicely matching raylib, plus a wrapper to make it easier for C# devs to use, as otherwise the C# dev needs to do this wrapper themselves.

```cs
[DllImport(nativeLibName, CallingConvention = CallingConvention.Cdecl)]
public static extern byte** GetDroppedFiles(int* count);

public static ReadOnlySpan<string> GetDroppedFiles()
{
    int count;
    var buffer = GetDroppedFiles(&count);
    var files  = new string[count];

    for (int i = 0; i < count; i++)
    {
        files[i] = Marshal.PtrToStringUTF8((IntPtr)buffer[i]);
    }

    return files;
}

```

### Environment

N/A
Win10 x64, dotnet6

**Comment by jasonswearingen** [comment]
as this is a breaking api change, it seems that now would be a good time to make it (prior to `raylib 4.0` release)

**Comment by DaZombieKiller** [comment]
> Not sure about returning ReadOnlySpan though.

Yeah, it might be better if it just returned a `string[]` directly. It would make it more obvious that it's an allocating API, and you can always assign the result to a `ReadOnlySpan<string>` yourself. Ideally, Raylib would have an additional `GetDroppedFilesCount` API, so you could preallocate a buffer yourself and have the API write into it (which would be translated into something like a `Span<string>` parameter in the C# binding, or something like `Span<Utf8CString>` where `Utf8CString` is just a `struct` wrapping a `byte*` if you wanted absolutely minimal allocations.)

**Comment by 9ParsonsB** [comment]
@jasonswearingen I am picking this up for the 4.0 release. We have already taken a look at string handling and the method you mention (see 4.0 PR #100).

Do you have any other recommendations or pain points in the API where we can look to improve?

**Comment by jasonswearingen** [comment]
I think that requiring the user to write glue code is the best "smell test".   so in your or the examples use of raylib, if there are areas that need similar workflows.    Though I don't think there are many like that.  I think the biggest pain was needing to cast everything to `IntPtr` which seems to be resolved in the 4.0 changes!
